### PR TITLE
Check return value of get_post_meta

### LINKED ISSUE #1107: `Utils\prepare_meta()` does not validate return value of `get_post_meta()`
### Describe the bug

The `prepare_meta()` utility function accepts a post ID and prepares the meta data for consumption/distribution.

https://github.com/10up/distributor/blob/ba8834c2a1e47689763e877c442a57b579546159/includes/utils.php#L420-L427

As [the return value of `get_post_meta()` is mixed](http://developer.wordpress.org/reference/functions/get_post_meta/), it is not guaranteed that `$meta` contains an empty array. Before using the data in a `foreach` loop, the variable should be validated.

### Steps to Reproduce

1. Enable debug logging in wp-config
   ```php
   define( 'WP_DEBUG', true );
   define( 'WP_DEBUG_LOG', true );
   define( 'WP_DEBUG_DISPLAY', false );
   ```
2. Run `wp shell` and call the function with an invalid post ID: `\Distributor\Utils\prepare_meta('bug');`
3. Open the `debug.log` file in your wp-content directory and observe the warning is thrown: `PHP Warning:  Invalid argument supplied for foreach() in .../includes/utils.php on line 427`

### Screenshots, screen recording, code snippet

Error log and stack trace:

```
[05-Jul-2023 03:36:38 UTC] PHP Warning:  Invalid argument supplied for foreach() in /vagrant/content/plugins/distributor/includes/utils.php on line 427
[05-Jul-2023 03:36:38 UTC] PHP Stack trace:
[05-Jul-2023 03:36:38 UTC] PHP   1. {main}() /usr/local/src/wp-cli/bin/wp:0
[05-Jul-2023 03:36:38 UTC] PHP   2. include() /usr/local/src/wp-cli/bin/wp:4
[05-Jul-2023 03:36:38 UTC] PHP   3. include() phar:///usr/local/src/wp-cli/bin/wp/php/boot-phar.php:11
[05-Jul-2023 03:36:38 UTC] PHP   4. WP_CLI\bootstrap() phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php:27
[05-Jul-2023 03:36:38 UTC] PHP   5. WP_CLI\Bootstrap\LaunchRunner->process($state = class WP_CLI\Bootstrap\BootstrapState { private $state = ['context_manager' => class WP_CLI\ContextManager { private $contexts = ['cli' => class WP_CLI\Context\Cli {  }, 'admin' => class WP_CLI\Context\Admin {  }, 'frontend' => class WP_CLI\Context\Frontend {  }, 'auto' => class WP_CLI\Context\Auto { private $context_manager = ... }]; private $current_context = 'cli' }] }) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/bootstrap.php:78
[05-Jul-2023 03:36:38 UTC] PHP   6. WP_CLI\Runner->start() phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php:28
[05-Jul-2023 03:36:38 UTC] PHP   7. WP_CLI\Runner->run_command_and_exit($help_exit_warning = *uninitialized*) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php:1237
[05-Jul-2023 03:36:38 UTC] PHP   8. WP_CLI\Runner->run_command($args = [0 => 'shell'], $assoc_args = [], $options = *uninitialized*) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php:440
[05-Jul-2023 03:36:38 UTC] PHP   9. WP_CLI\Dispatcher\Subcommand->invoke($args = [], $assoc_args = [], $extra_args = []) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php:417
[05-Jul-2023 03:36:38 UTC] PHP  10. call_user_func:{phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php:491}($function_name = class Closure { virtual $closure = "$this->WP_CLI\Dispatcher\{closure}", public $static = ['callable' => [0 => 'Shell_Command', 1 => '__invoke']]; public $parameter = ['$args' => '<required>', '$assoc_args' => '<required>'] }, ...$parameters = variadic([])) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php:491
[05-Jul-2023 03:36:38 UTC] PHP  11. WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure:phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php:97-104}($args = [], $assoc_args = []) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php:491
[05-Jul-2023 03:36:38 UTC] PHP  12. call_user_func:{phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php:100}($function_name = [0 => class Shell_Command {  }, 1 => '__invoke'], ...$parameters = variadic([])) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php:100
[05-Jul-2023 03:36:38 UTC] PHP  13. Shell_Command->__invoke($_ = [], $assoc_args = []) phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php:100
[05-Jul-2023 03:36:38 UTC] PHP  14. WP_CLI\Shell\REPL->start() phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/shell-command/src/Shell_Command.php:52
[05-Jul-2023 03:36:38 UTC] PHP  15. eval() phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:46
[05-Jul-2023 03:36:38 UTC] PHP  16. Distributor\Utils\prepare_meta($post_id = 'bob') phar:///usr/local/src/wp-cli/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php(46) : eval()'d code:1

```

### Environment information

_No response_

### WordPress information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by dhanendran**
### Description of the Change
Validate the return value of get_post_meta and make sure it is array.

Closes #1107

### How to test the Change

1. Enable debug logging in wp-config
```
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
define( 'WP_DEBUG_DISPLAY', false );
```
2. Run `wp shell` and call the function with an invalid post ID: `\Distributor\Utils\prepare_meta('bug');`
3. Open the `debug.log` file in your `wp-content` directory and observe the no warning is thrown: `PHP Warning:  Invalid argument supplied for foreach() in .../includes/utils.php on line 427`

### Credits
Props @dhanendran 

### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: includes/utils.php [hunk 1]
```diff
@@ -435,6 +435,7 @@ function excluded_meta() {
 function prepare_meta( $post_id ) {
 	update_postmeta_cache( array( $post_id ) );
 	$meta          = get_post_meta( $post_id );
+	$meta          = is_array( $meta ) ? $meta : array();
```
Comment (faisal-alvi) [inline includes/utils.php:443]: Here are the chances of a PHP 8+ incompatibility warning: `Automatic conversion of false to array is deprecated`.

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
@dhanendran Thanks for the PR. As [the return value of get_post_meta() is mixed](http://developer.wordpress.org/reference/functions/get_post_meta/), it may return `false` and we will end up overwriting it with an `array()` which may throw the PHP warning in v8+.

File: includes/utils.php (hunk 1 above)
Comment (kirtangajjar) [reply includes/utils.php:443]: @faisal-alvi I've added early return in case meta is false.

File: includes/utils.php [hunk 2]
```diff
@@ -435,6 +435,12 @@ function excluded_meta() {
 function prepare_meta( $post_id ) {
 	update_postmeta_cache( array( $post_id ) );
 	$meta          = get_post_meta( $post_id );
```
Comment (faisal-alvi) [inline includes/utils.php]:
```suggestion
	$meta = get_post_meta( $post_id );
```

File: includes/utils.php [hunk 3]
```diff
@@ -435,6 +435,12 @@ function excluded_meta() {
...
+
+	if ( false === $meta ) {
+		return array();
+	}
+
+	$meta          = is_array( $meta ) ? $meta : array();
```
Comment (faisal-alvi) [inline includes/utils.php:443]:
```suggestion
	$meta = is_array( $meta ) ? $meta : array();
```

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
@kirtangajjar thanks for the fix. A couple of minor PHPCS issues are noted below.

An early return would work but will never reach the `dt_prepared_meta` filter. We can consider it as an exception though (i.e. empty meta would not reach the filter).

**Comment by kirtangajjar** [comment]
@faisal-alvi Made the requested changes. LMk if there are any more

File: includes/utils.php [hunk 4]
```diff
@@ -434,7 +434,13 @@ function excluded_meta() {
...
-	$meta          = get_post_meta( $post_id );
+	$meta = get_post_meta( $post_id );
+
+	if ( false === $meta ) {
+		return array();
+	}
+
+	$meta = is_array( $meta ) ? $meta : array();
 	$prepared_meta = array();
 	$excluded_meta = excluded_meta();
```
Comment (faisal-alvi) [inline includes/utils.php]:
```suggestion
	$meta          = is_array( $meta ) ? $meta : array();
	$prepared_meta = array();
	$excluded_meta = excluded_meta();
```

**Review by faisal-alvi — APPROVED** [review]
Ignoring E2E failures as they are being addressed in #1242

**Pull request merged**